src/core/projectinfo.cpp

```diff
@@ -413,14 +453,58 @@ void ProjectInfo::restoreSettings( QString &projectFilePath, QgsProject *project
     mapCollection.applyTheme( QStringLiteral( "::QFieldLayerTreeState" ), layerTree->layerTreeModel()->rootGroup(), layerTree->layerTreeModel() );
   }
 
-  const QString snappingConfig = settings.value( QStringLiteral( "/qgis/projectInfo/%1/snappingconfig" ).arg( projectFilePath ), QString() ).toString();
-  if ( !snappingConfig.isEmpty() )
+  settings.beginGroup( QStringLiteral( "/qgis/projectInfo/%1/layerSnapping" ).arg( projectFilePath ) );
+  const QStringList values = settings.allKeys();
+  if ( !values.isEmpty() )
   {
-    QDomDocument document;
-    document.setContent( snappingConfig );
+    QgsSnappingConfig config = project->snappingConfig();
+    if ( values.contains( QStringLiteral( "enabled" ) ) )
+    {
+      config.setEnabled( settings.value( QStringLiteral( "enabled" ), true ).toBool() );
+    }
+
+    ids = settings.childGroups();
+    if ( !ids.isEmpty() )
+    {
+      const bool isDataset = project->readBoolEntry( QStringLiteral( "QField" ), QStringLiteral( "isDataset" ), false );
+      const QList<QgsMapLayer *> mapLayers = isDataset ? project->layerStore()->mapLayers().values() : QList<QgsMapLayer *>();
+
+      for ( QString id : ids )
```

Comment:
```suggestion
      for ( QString id : std::as_const( ids ) )
```

... or this if you want to recycle the `ids` var instead of making it `const`

Reply:
@m-kuhn , I'll do that sir :)